src/main/java/io/javalin/core/JavalinServlet.kt

```diff
@@ -6,36 +6,29 @@
 
 package io.javalin.core
 
-import io.javalin.*
+import io.javalin.Context
+import io.javalin.Handler
+import io.javalin.MethodNotAllowedResponse
+import io.javalin.NotFoundResponse
 import io.javalin.core.util.*
-import io.javalin.security.AccessManager
 import io.javalin.security.CoreRoles
 import io.javalin.security.Role
-import io.javalin.security.SecurityUtil
-import io.javalin.staticfiles.ResourceHandler
 import java.io.InputStream
 import java.util.zip.GZIPOutputStream
 import javax.servlet.http.HttpServletRequest
 import javax.servlet.http.HttpServletResponse
 
 class JavalinServlet(private val appAttributes: Map<Class<*>, Any>) {
 
+    val config = JavalinServletConfig(this)
+
     val matcher = PathMatcher()
     val exceptionMapper = ExceptionMapper()
     val errorMapper = ErrorMapper()
-    val singlePageHandler = SinglePageHandler()
-    var resourceHandler: ResourceHandler? = null
-    var accessManager = AccessManager { handler, ctx, permittedRoles -> SecurityUtil.noopAccessManager(handler, ctx, permittedRoles) }
-    var requestLogger: RequestLogger? = null
-    var dynamicGzipEnabled = true
-    var autogeneratedEtagsEnabled = false
-    var defaultContentType = "text/plain"
-    var maxRequestCacheBodySize = 4096L
-    var prefer405over404 = false
 
     fun service(servletRequest: HttpServletRequest, res: HttpServletResponse) {
```

Comment:
Just an idea worth considering for the future:

As `JavalinServlet.service()` method grows, we could benefits from clear structure in here, where we have ordered `before`, `user handler` and `after` logic, where before/after is internal or users  ```Handler { before(ctx) {...}, after(ctx) {...} }``` kind of handler. Basically it resembles decorators or filters.

This kind of separation would help us by keeping some of those handlers could be added on the time of applying JavalinServletConfig.

examples:
```
internalPipeline.forEach { step -> step.before(ctx); registeredHandlerByUser.accept(ctx); step.after(ctx) }
```

Alternativelly we could consider servlet filter chain (where each filter/handler manipulates (req,res) and passes them to next handler in the chain):
```
securityFilter->loggingFilter->customFilter->req handler resp->customFilter->loggingFilter->securityFilter
```

Reply:
I find the current setup to be very easy to work with, but I'm open to suggestions. Maybe you could make a PR, or at least a rough draft?

Reply:
Yes, let's try it some time later. It's not related to this pull request so we can take it offline!